### PR TITLE
Adding test to test Ephemerons in the Old Space

### PR DESCRIPTION
**Pull request by jordanmontt**
Added a class that tests some edge cases of the finalization algorithm. The class has failing tests which is good. That means that we found some errors on the finalization algorithm. Next step is to fix those errors

### CODE REVIEW

**Comment by guillep** [comment]
Hi @jordanmontt , is this finally an issue or not?

File: smalltalksrc/VMMakerTests/VMSpurEphemeronsAlgorithmTest.class.st [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+	self keepObjectInVMVariable1: ephemeronObjectOopOne.
+	self keepObjectInVMVariable2: ephemeronObjectOopTwo.
+
+	"Collect ephemerons"
+	memory doScavenge: 1.
+
+	self assert: ({ ephemeronObjectOopOne. ephemeronObjectOopTwo } includes: memory dequeueMourner).
+	self assert: ({ ephemeronObjectOopOne. ephemeronObjectOopTwo } includes: memory dequeueMourner)
+]
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMSpurEphemeronsAlgorithmTest.class.st]: This test will fail so far by design.

Ephemerons are processed one-by-one during the scavenge.
When the first ephemeron is processed, the key is marked to survive (copied), and suddenly the second ephemeron's key survived and should not be copied.

There is a test, with the same setup contradicting your example

`testScavengeTwoEphemeronObjectsReferencingSameDyingObjectsShouldAddOnlyOneEphemeron`

This will force only one ephemeron to be finalized per cycle, which is indeed not optimal.
But it should be worked separately.

File: smalltalksrc/VMMakerTests/VMSpurEphemeronsAlgorithmTest.class.st [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+	#name : #VMSpurEphemeronsAlgorithmTest,
+	#superclass : #VMSpurInitializedOldSpaceTest,
+	#instVars : [
+		'ephemeronObjectOopOne',
+		'ephemeronObjectOopTwo',
+		'nonEphemeronObjectOopOne',
+		'nonEphemeronObjectOopTwo'
+	],
+	#category : #'VMMakerTests-MemoryTests'
+}
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMSpurEphemeronsAlgorithmTest.class.st]: Why a new class when there already was `VMSpurScavengeEphemeronTest`?

File: smalltalksrc/VMMakerTests/VMSpurEphemeronsAlgorithmTest.class.st [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+	self keepObjectInVMVariable1: ephemeronObjectOopOne.
+	self keepObjectInVMVariable2: ephemeronObjectOopTwo.
+
+	"Collect ephemerons"
+	memory fullGC.
+
+	self assert: ({ ephemeronObjectOopOne. ephemeronObjectOopTwo } includes: memory dequeueMourner).
+	self assert: ({ ephemeronObjectOopOne. ephemeronObjectOopTwo } includes: memory dequeueMourner)
+]
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMSpurEphemeronsAlgorithmTest.class.st]: This test is indeed missing in our suite, but this should be added to `VMSpurOldSpaceGarbageCollectorTest`?

**Review by guillep — CHANGES_REQUESTED** [review]
I don't think we should integrate it as it is. Besides the structural silly change of moving the tests to the right classes, there is at least one test that contradicts another tests that already exist. So one of them is wrong, we cannot keep both.

Also, adding broken tests to the suite is not good. They should come with the proper fix...

**Comment by jordanmontt** [comment]
Thanks for the feedback! I will move the useful tests to the default test class and close this PR

**Comment by jordanmontt** [comment]
Mmmm the new tonel format :/ *$%$#@#$

**Comment by jordanmontt** [comment]
It looks like it's ready to be integrated @guillep . Failing tests Tests-Linux-x86_64-CoInterpreter Tests-Linux-x86_64-CoInterpreter-worker don't seem related
